src/rm_tdigest.c

```diff
@@ -915,31 +915,49 @@ size_t TDigestMemUsage(const void *value) {
 
 int TDigestModule_onLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     // TODO: add option to set defaults from command line and in program
-    RedisModuleTypeMethods tm = {.version = REDISMODULE_TYPE_METHOD_VERSION,
-                                 .rdb_load = TDigestRdbLoad,
-                                 .rdb_save = TDigestRdbSave,
-                                 .aof_rewrite = RMUtil_DefaultAofRewrite,
-                                 .mem_usage = TDigestMemUsage,
-                                 .free = TDigestFree,
-                                 .defrag = TDigestDefrag};
+    RedisModuleTypeMethods tm = {
+        .version = REDISMODULE_TYPE_METHOD_VERSION,
+        .rdb_load = TDigestRdbLoad,
+        .rdb_save = TDigestRdbSave,
+        .aof_rewrite = RMUtil_DefaultAofRewrite,
+        .mem_usage = TDigestMemUsage,
+        .free = TDigestFree,
+        .defrag = TDigestDefrag,
+    };
 
     TDigestSketchType = RedisModule_CreateDataType(ctx, "TDIS-TYPE", TDIGEST_ENC_VER, &tm);
-    if (TDigestSketchType == NULL)
+    if (TDigestSketchType == NULL) {
         return REDISMODULE_ERR;
+    }
+
+#define MODULE_ACL_CATEGORY_NAME "tdigest"
+    RegisterAclCategory(ctx);
+    RegisterCommandWithModesAndAcls(ctx, "tdigest.create", TDigestSketch_Create, "write deny-oom",
```

Comment:
I know the ACL category name and the command name aren't the same, but here it all looks like it could have been a string concatenation of `MODULE_ACL_CATEGORY_NAME` + `".create"`. I do not insist on the change, though.

Reply:
i thought about it, but there's other acl category names where it's not the case. eg bloom != bf

Reply:
Okay :(